modules/onnx/1.15.0/patches/add_build_file.patch

```diff
@@ -0,0 +1,106 @@
+diff --git BUILD.bazel BUILD.bazel
+new file mode 100644
+index 00000000..b94cab17
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,100 @@
++load("@rules_proto//proto:defs.bzl", "proto_library")
++load("@rules_cc//cc:defs.bzl", "cc_library", "cc_proto_library")
++load("@rules_python//python:defs.bzl", "py_binary")
++
++py_binary(
++    name = "gen_proto",
++    srcs = ["onnx/gen_proto.py"],
++    data = [
++        "onnx/onnx.in.proto",
++        "onnx/onnx-operators.in.proto",
++        "onnx/onnx-data.in.proto",
++    ],
++)
++
++genrule(
++    name = "generate_onnx_proto",
++    outs = [
++        "onnx/onnx-ml.proto",
++    ],
++    cmd = "$(location :gen_proto) -p onnx -o $(@D) onnx -m",
++    tools = [":gen_proto"],
++)
++
++genrule(
++    name = "generate_onnx_operators_proto",
++    outs = [
++        "onnx/onnx-operators-ml.proto",
++    ],
++    cmd = "$(location :gen_proto) -p onnx -o $(@D) onnx-operators -m >/dev/null",
++    tools = [":gen_proto"],
++)
++
++genrule(
++    name = "generate_onnx_data_proto",
++    outs = [
++        "onnx/onnx-data.proto",
++    ],
++    cmd = "$(location :gen_proto) -p onnx -o $(@D) onnx-data -m >/dev/null",
++    tools = [":gen_proto"],
++)
++
++cc_library(
++    name = "onnx",
++    srcs = glob(
++        [
++            "onnx/*.cc",
++            "onnx/common/*.cc",
++            "onnx/defs/**/*.cc",
++            "onnx/shape_inference/*.cc",
++            "onnx/version_converter/*.cc",
++        ],
++        exclude = [
++            "onnx/cpp2py_export.cc",
++        ],
++    ),
++    hdrs = glob([
++        "onnx/*.h",
++        "onnx/version_converter/*.h",
++        "onnx/common/*.h",
++        "onnx/defs/**/*.h",
++        "onnx/shape_inference/*.h",
++        "onnx/version_converter/adapters/*.h",
++    ]),
++    defines = [
++        "ONNX_ML=1",
++        "ONNX_NAMESPACE=onnx",
++    ],
++    includes = [
++        "onnx",
++    ],
++    visibility = ["//visibility:public"],
++    deps = [
++        ":onnx_proto_lib",
++    ],
++)
++
++cc_library(
++    name = "onnx_proto_headers",
++    hdrs = glob([
++        "onnx/*_pb.h",
++    ]),
++    visibility = ["//visibility:public"],
```

Comment:
Is it intentional that these header files have a publicly visible target, but the actual `cc_proto_library` doesn't? (I know nothing about use cases, so this may very well be entirely correct)

Reply:
make onnx proto lib public [here](https://github.com/bazelbuild/bazel-central-registry/pull/1525/commits/1e8ca7ef48b6584327158f69a9496b93c39ee071)